ios/testmanagerd/testlistener.go

```diff
@@ -164,7 +164,31 @@ func (t *TestListener) testSuiteDidStart(suiteName string, date string) {
 }
 
 func (t *TestListener) testCaseDidStartForClass(testClass string, testMethod string) {
+	// Find the existing test suite or create a new one if not found
 	ts := t.findTestSuite(testClass)
+	if ts == nil {
+		// If no test suite is found and we're not in a running test suite,
+		// we should use the runningTestSuite instead of creating a new one.
+		// This handles cases where testCaseDidStartForClass is called before
+		// testSuiteDidStart, which would otherwise result in a nil pointer dereference.
```

Comment:
> This handles cases where testCaseDidStartForClass is called before testSuiteDidStart, which would otherwise result in a nil pointer dereference.

Are you sure that's happening and the call for testSuiteDidStart` after `testCaseDidStartForClass` is not for a different test suite? It sounds very strange

Reply:
I meant reproducing on an actual device with a real test execution